WebInstall/explainer_cross_domain.md

```diff
@@ -0,0 +1,259 @@
+
+A website that wants to install apps will require this new permission and will only be able to prompt the user for this in a period of time defined by the implementer. This will avoid spam from websites constantly asking for a permission to install apps, and will force websites to only prompt when there is a meaningful user intent to install apps.
+
+The installation permission for an origin should be time-limited and expire after a period of time defined by the UA. After the permission expires the UA will prompt again for permission from the user.
+
+####  **Integration with the Permissions API**
+A new permission that can be associated with an origin means a new integration with the [Permissions API](https://www.w3.org/TR/permissions/). The install API will make available the "installation" [PermissionDescriptor](https://www.w3.org/TR/permissions/#dom-permissiondescriptor) as a new [*powerful feature*](https://www.w3.org/TR/permissions/#dfn-specifies-a-powerful-feature). This would make it possible to know programmatically if `install` would be blocked.
+
+```javascript
+/* example of querying for the state of an installation permission using the Permission API  */
+
+const { state } = await navigator.permissions.query({
+  name: "installation"
+});
+switch (state) {
+  case "granted":
+    navigator.install('https://elk.zone');
+    break;
+  case "prompt":
+    //shows the install button in the web UI
+    showInstallButton();
+    break;
+  case "denied":
+    redirectToAppStore();
+    break;
+}
+```
+####  **Install Sources manifest field**
+* A new field called `install_sources` will be added to the manifest file to have a control list of sites that can install the app. In its most restrictive case, the developer can specify to not allow installation from any other origin, in which case the PWA conforms to its usual behaviour of only being able to be installed from its same origin.
+
+    * **`getInstalledApps()` method: If supported by the UA, the `getInstalledApps` method returns a list of the content that has been installed from that installation origin.** This means the installation origin will be able to know which applications it has installed, until cache is cleared. The installation origin *will not* be informed of any apps installed by other means, whether via another installation origin, directly through the browser, or by a native app store. The method returns a list of manifest ids of content installed from the calling origin. Additionally, if the browser has an active 'Do Not Track (DNT)', equivalent 'Global Privacy Control (GPC)' setting, is in Private browsing mode, or is an opinionated browser towards privacy, this is ignored and installation origins will not be allowed to know if that application is installed.
```

Comment:
Is this on `navigator`?  Is it `async`? In scenarios where it has installed nothing or DNT/GPC/etc... is in play, does this return back `null` or an empty list?